config/config.go

```diff
@@ -746,6 +748,32 @@ func (re *Regexp) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// Matchers encapsulates labels.Matchers and makes it YAML marshalable.
+type Matchers labels.Matchers
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface for Matchers.
+func (m *Matchers) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	var lines []string
+	if err := unmarshal(&lines); err != nil {
+		return err
+	}
+	for _, line := range lines {
+		labelMatchers, err := labels.ParseMatchers(line)
+		if err != nil {
+			return err
+		}
+		for _, matcher := range labelMatchers {
+			matchers, err := labels.NewMatcher(matcher.Type, matcher.Name, matcher.Value)
+			if err != nil {
+				return err
+			}
+			*m = append(*m, matchers)
+		}
+	}
+	return nil
```

Comment:
I guess it cannot harm to do a `sort.Sort(*m)` before returning for consistency. (And later, once the deprecated matchers are gone, we don't need to sort in `dispatch/route.go` again.)

Reply:
Yes, sounds good